main.go

```diff
@@ -413,6 +428,15 @@ func (o *Options) runServerMode() error {
 		return err
 	}
 
+	var sippyserverPinnedTime time.Time
+
+	// the pinnedDateTime that is passed in is always set
+	// so we have to check the input parameter to verify if
+	// we want to pass it along or provide the empty sippyserverPinnedTime
+	if len(o.PinnedDateTime) > 0 {
+		sippyserverPinnedTime = pinnedDateTime
+	}
```

Comment:
As far as I can tell, you always pass a value in for o.PinnedDateTime that's > 0

Reply:
That is the input parameter value, if it isn't provided then the len(..) should be 0, seems to work but can review more

Reply:
![image](https://user-images.githubusercontent.com/36795216/187798511-7daed67f-9c50-4bc6-b926-9b199e0495ae.png)

It could be confusing with me passing in the none empty pinnedDateTime and then referencing the string option value but that was because I needed the value for the db.New call but wanted to leave the server value unpinned when the option wasn't specified.

Reply:
I see, I think I was confusing o.PinnedDateTime vs. the function parameter pinnedDateTime.  It's just a little confusing, since you ALWAYS pass a value for the time.Time function parameter.  Why do you need to pass it if a user didn't provide o.PinnedDateTime?

Reply:
The materialized views always need a dateTime now (no longer use NOW in the sql) (we can discuss this approach) in main.Run we need the dateTime where the dbs are created, there is also a call in runServerMode that calls db.New if needed, so I didn't want to parse the input over and over (but could) however that left me needing to flag when the server needed to be passed a specific dateTime (pinned) vs when it should get an empty one so the server knows to call time.Now() in that case.

Reply:
Why not pass a `*time.Time` and check for nil? Use time.Now if it's nil, otherwise use the pinned value?

Reply:
Yes, I think it's important to be more clear in the code about what's going on.

If there's an argument called pinnedDateTime, it's reasonable for a developer to assume if it has a non-zero value then there's something pinned.  That's not the case here, you always pass in a non-zero value -- and a user has to backtrack and look at the original raw options string to figure it out.  

You can hold all that state clearly by making time.Time a pointer and checking for nil, and if nil using `time.Now().`

I'm also not convinced passing a date to db.New is the right solution. I would personally separate the concerns of creating a client, and syncing the schema/generating mat views/etc.

Reply:
Latest version is much better

Reply:
Agree, thanks for the nudge :-)